### PR TITLE
New Customizable Watchface wip

### PR DESCRIPTION
**Pull request by Philoul**
This new watchface (default face is an Analog watchface close to AAPS v2) can be customized from an external file (zip file that includes a json file to customize all the views included into the watchface, including a bitmap background...)
All availables watchfaces should be put into AAPS/exports folder
Export Watchface button can export (into exports folder) the custom watchface included into your Wear Watch (you can have that way a template to not start from scratch if you want to create a new watchface)
Load Watchface button is for selection and updload to watch a new watchface
Send Watchface button can send again selected watchface
Default Watchface reset custom watchface to default one

You can tune background with a dedicated bitmap (jpg or png format) within a file named Background.jpg (or Background.png) into zip file (see my custom watchface as example)
Michel Herbelin example : [Michel Herbelin.zip](https://github.com/nightscout/AndroidAPS/files/12279303/Michel.Herbelin.zip)
Simple Digital (designed by @jbr7rr ) : [SimpleDigital.zip](https://github.com/nightscout/AndroidAPS/files/12317569/SimpleDigital.zip)

image

Mandatory information to have a "valid" zip file are:
- A valid JSON file (named CustomWatchface.json) that will contains all information concerning the watchface, view visibles, position, font, style, colors ...)
- A bitmap named CustomWatchface.png (or CustomWatchface.jpg)  for Watchface selection

Optional informations are:
- Background.jpg (or Background.png) file
- CoverChart.png file
- CoverPlate.png file
- HourHand,png, MinuteHand.png, SecondHand.png

Be carefull, zip file must not be too big (400 or 500kb is a maximum, so prefer zip file for background and small size png file for other bitmap)

Concerning json file, all position, and size are in pixel based on a 400px x 400px size (width, height, topmargin, leftmargin and textsize)
all colors should be in "#RRGGBB" format (for ex "#FFFFFF" for White)

### CODE REVIEW

File: app-wear-shared/shared/src/main/java/info/nightscout/rx/weardata/EventData.kt [hunk 1]
```diff
@@ -13,13 +14,20 @@ sealed class EventData : Event() {
 
     fun serialize() = Json.encodeToString(serializer(), this)
 
+    fun serializeByte() = ProtoBuf.encodeToByteArray(serializer(), this)
```
Comment (jbr7rr) [inline app-wear-shared/shared/src/main/java/info/nightscout/rx/weardata/EventData.kt:17]: Gives warning during build

File: app-wear-shared/shared/src/main/java/info/nightscout/rx/weardata/EventData.kt [hunk 2]
```diff
@@ -13,13 +14,20 @@ sealed class EventData : Event() {
...
     companion object {
 
         fun deserialize(json: String) = try {
             Json.decodeFromString(serializer(), json)
         } catch (ignored: Exception) {
             Error(System.currentTimeMillis())
         }
+
+        fun deserializeByte(byteArray: ByteArray) = try {
+            ProtoBuf.decodeFromByteArray(serializer(), byteArray)
```
Comment (jbr7rr) [inline app-wear-shared/shared/src/main/java/info/nightscout/rx/weardata/EventData.kt:27]: Same, gives warning during build

Comment (Philoul) [reply app-wear-shared/shared/src/main/java/info/nightscout/rx/weardata/EventData.kt:27]: Yes the warning arrived one or 2 days ago when I updated Android Studio and my working branch with latest dev 😥...
Itried to fix it with suggested proposal, but watch face was no more sent to the watch (I think we have to manually accept something somewhere...)
I couldn't use standard json serialuzer (Size of json serialization is between 3 and 4 times greater, so about 1,7mb for my test watchface, butthe threshold to be able to send data to watch is at about 1mB...
With protobuf serializer, size is reduced to about 400/500kb and can be sent (as bytearray) to the watch...
I will try to see if I can fix it (with an authorization request, or by using another binary serialyser...)

**Comment by Philoul** [comment]
I made an important refactoring of the code.
Warning is removed, and now watch displayed on Wear Plugin (in AAPS) is the one downloaded from the watch
=> You must have a Wear watch connected to be able to load a new watchface
=> If you already started to test this PR, you have to update AAPS and Wear together to be able to load/export watchfaces

**Comment by Philoul** [comment]
See below a watchface proposed by @jbr7rr 
[SimpleDigital.zip](https://github.com/nightscout/AndroidAPS/files/12317523/SimpleDigital.zip)
CustomWatchface
I like the idea to build and share new watchfaces with all users...
image

**Pull request merged**